service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -228,10 +236,18 @@ public ResponseEntity<ApiWorkspaceDescription> updateWorkspace(
     }
     workspaceService.validateWorkspaceAndAction(
         userRequest, workspaceUuid, SamConstants.SamWorkspaceAction.WRITE);
+    var userStatusInfo =
+        SamRethrow.onInterrupted(
+            () -> samService.getUserStatusInfo(userRequest),
```

Comment:
You could just move the `SamRethrow` phrase into a method in SamService; something like
`getUserStatusInfoWithRethrow()`

I would prefer pushing the whole thing into the service instead of repeatedly doing the Sam call and passing the results as params.

Reply:
I thought about extract a method. It seems to trivial/specific to me at the time. 

i'm not sure I understand this part : 
`I would prefer pushing the whole thing into the service instead of repeatedly doing the Sam call and passing the results as params.`

just to clarify, pushing the whole thing as in this entire block?

```
var userStatusInfo =
   SamRethrow.onInterrupted(
      () -> samService.getUserStatusInfo(userRequest),
      ...);
Workspace workspace =
        workspaceService.updateWorkspace(
            workspaceUuid, body.getUserFacingId(), body.getDisplayName(), body.getDescription());

            workspaceUuid,
            body.getUserFacingId(),
            body.getDisplayName(),
            body.getDescription(),
            userStatusInfo.getUserEmail(),
            userStatusInfo.getUserSubjectId());
    WsmIamRole highestRole = workspaceService.getHighestRole(workspaceUuid, userRequest);
    ApiWorkspaceDescription desc = buildWorkspaceDescription(workspace, highestRole);
```

Reply:
i can have a WorkspaceActivitylogService that has a logActivity(userRequest, ...) that can dry out some code
```
var userStatusInfo =
        samService.getUserStatusInfoWithRethrow(new AuthenticatedUserRequest());
    workspaceActivityLogDao.writeActivity(
        uuid,
        new DbWorkspaceActivityLog()
            .operationType(OperationType.GRANT_WORKSPACE_ROLE)
            .userEmail(userStatusInfo.getUserEmail())
            .subjectId(userStatusInfo.getUserSubjectId()));
```